pf/tfbridge/provider_checkconfig.go

```diff
@@ -48,35 +57,22 @@ func (p *provider) CheckConfigWithContext(
 		ProviderConfig: inputs,
 	})
 
+	if !news.DeepEquals(inputs) {
```

Comment:
This is not safe on arbitrary values, as the tests indicate.

This is why https://github.com/pulumi/pulumi-go-provider/blob/b60247b0e97434ac7be8b9a344deb4cbad524daa/infer/internal/ende/util.go#L95-L104 exists.

Reply:
Ouch. Very well, good catch.